cloudferrylib/os/image/glance_image.py

```diff
@@ -105,24 +139,55 @@ def convert(glance_image, cloud):
         """
 
         resource = cloud.resources[utl.IMAGE_RESOURCE]
+        keystone = cloud.resources["identity"]
         gl_image = {
-            'id': glance_image.id,
-            'size': glance_image.size,
-            'name': glance_image.name,
-            'checksum': glance_image.checksum,
-            'container_format': glance_image.container_format,
-            'disk_format': glance_image.disk_format,
-            'is_public': glance_image.is_public,
-            'protected': glance_image.protected,
-            'resource': resource,
-            'properties': ({
-                'image_type': glance_image.properties['image_type']}
-                if 'image_type' in glance_image.properties
-                else glance_image.properties)
-        }
-
+            k: w for k, w in glance_image.to_dict().items(
+            ) if k in CREATE_PARAMS}
+        # we need to pass resource to destination to copy image
+        gl_image.update({'resource': resource})
+        # at this point we write name of owner of this tenant
+        # to map it to different tenant id on destination
+        gl_image.update(
+            {'owner_name': keystone.get_tenant_by_id(
+                glance_image.owner).name})
+        if resource.is_snaphot(glance_image):
+            # for snapshots we need to write snapshot username to namespace
+            # to map it later to new user id
+            user_ids = [i.id for i in keystone.keystone_client.users.list()]
+            user_id = gl_image["properties"].get("user_id")
+            if user_id in user_ids:
+                gl_image["properties"]["user_name"] = \
+                    keystone.keystone_client.users.get(user_id).name
         return gl_image
 
+    def is_snaphot(self, img):
+        # snapshots have "properties" field
+        return img.to_dict().get("properties")
+
+    def get_tags(self):
```

Comment:
this is the stub to implement this functionality after we have v2 client support